app/src/middleware/authorization.js

```diff
@@ -207,6 +210,29 @@ const hasPermission = (permission) => {
   };
 };
 
+/**
+ * if non-IDIR user, require userId, email or identityId query parameter
+ * route validationrequires a valid email
+ */
+const restrictNonIdirUserSearch = async (req, _res, next) => {
```

Comment:
I'm now thinking this is the wrong thing to restrict.
It should be, 'when searching for non-idir users, require an exact match on either userId, email or identityId